moveit_core/CMakeLists.txt

```diff
@@ -21,8 +21,19 @@ endif()
 
 find_package(Boost REQUIRED system filesystem date_time thread iostreams)
 find_package(Eigen3 REQUIRED)
- # TODO: Move collision detection into separate packages
-find_package(Bullet REQUIRED)
+
+# TODO: Move collision detection into separate packages
+find_package(Bullet CONFIG)
+if(BULLET_FOUND AND BULLET_VERSION_STRING VERSION_GREATER "2.86")
+  set(BULLET_ENABLE "BULLET")
+  set(BULLET_INCLUDE_DIRS "${BULLET_ROOT_DIR}/${BULLET_INCLUDE_DIRS}")
+  set(BULLET_INCLUDE_DIR "${BULLET_ROOT_DIR}/${BULLET_INCLUDE_DIR}")
+  set(BULLET_LIBRARY_DIRS "${BULLET_ROOT_DIR}/${BULLET_LIBRARY_DIRS}")
+  set(BULLET_LIBRARIES "libBulletDynamics.so;libBulletCollision.so;libLinearMath.so;libBulletSoftBody.so")
```

Comment:
Can you explain why you added this?

Especially the `BULLET_LIBRARIES` values seem strange, as CMake typically uses absolute paths to libraries, and `BULLET_LIBRARIES` should already contain the proper values.

Reply:
Somehow, the `BulletConfig.cmake` file coming with `libbullet-dev` does not fit the library files coming with it. I was also wondering why this is the case. See the content of the config:

```
#                                               -*- cmake -*-
#
#  BulletConfig.cmake(.in)
#

# Use the following variables to compile and link against Bullet:
#  BULLET_FOUND              - True if Bullet was found on your system
#  BULLET_USE_FILE           - The file making Bullet usable
#  BULLET_DEFINITIONS        - Definitions needed to build with Bullet
#  BULLET_INCLUDE_DIR        - Directory where Bullet-C-Api.h can be found
#  BULLET_INCLUDE_DIRS       - List of directories of Bullet and it's dependencies
#  BULLET_LIBRARIES          - List of libraries to link against Bullet library
#  BULLET_LIBRARY_DIRS       - List of directories containing Bullet' libraries
#  BULLET_ROOT_DIR           - The base directory of Bullet
#  BULLET_VERSION_STRING     - A human-readable string containing the version

set ( BULLET_FOUND 1 )
set ( BULLET_USE_FILE     "lib/x86_64-linux-gnu/cmake/bullet/UseBullet.cmake" )
set ( BULLET_DEFINITIONS  "" )
set ( BULLET_INCLUDE_DIR  "include/bullet" )
set ( BULLET_INCLUDE_DIRS "include/bullet" )
set ( BULLET_LIBRARIES    "LinearMath;Bullet3Common;BulletInverseDynamics;BulletCollision;BulletDynamics;BulletSoftBody" )
set ( BULLET_LIBRARY_DIRS "lib/x86_64-linux-gnu" )
set ( BULLET_ROOT_DIR     "/usr" )
set ( BULLET_VERSION_STRING "2.87" )
```

And see the files coming with it at https://packages.ubuntu.com/bionic/i386/libbullet-dev/filelist. There is no `BulletInverseDynamics` coming with this package. I am also not very happy about the fact that I have to define them manually and do this concatenation with the root directory.. any ideas?

Reply:
When I simply use `find_package(Bullet)` this works fine, as it uses the `FindBullet.cmake`.

Reply:
I don't like this approach as well. What about providing our own FindBullet.cmake?

Reply:
Isn't this file provided by the cmake installation?

Reply:
Yes, cmake provides such a file, but we can overrule this default by providing our own, can't we?

Reply:
Any proposals where in the source tree such a file should go?

Reply:
In `moveit_core/CMakeModules`? In `moveit_core/CMakeLists.txt` you'd add something like this:
```
list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")

find_package(Bullet)
```

Reply:
I am struggling to get this running. My idea is to have the custom `FindBullet.cmake` be more or less a replicate of the [official one](https://github.com/Kitware/CMake/blob/master/Modules/FindBullet.cmake) plus checking somehow the `BulletConfig.cmake` which comes with `libbullet-dev` for the `BULLET_VERSION_STRING`. 

However if I do `include(.../BulletConfig.cmake)` in the `FindBullet.cmake` I get all the variables set which clutters the `FindBullet.cmake` to the point that I cannot build succesfully anymore. I would like to only partially include the `BulletConfig.cmake` but I cannot find that option in cmake. 

Any ideas?

Reply:
You can also use the cmake PkgConfig module (since bullet also has a bullet.pc file). Your FindBullet.cmake might then look something like this:
```
include(FindPackageHandleStandardArgs)
find_package(PkgConfig)
if(PKGCONFIG_FOUND)
    pkg_check_modules(bullet>=2.88)
endif()
find_package_handle_standard_args(Bullet DEFAULT_MSG BULLET_LIBRARIES BULLET_INCLUDE_DIRS)
```

Reply:
That worked :), thank you @mamoll for the help!